## Statistic Test
#### Prove that labeled data is different by runining ANOVA + T test or kruskall wallis test.

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv("dataset_limpio.csv")

data.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [3]:
import scipy.stats as stats
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

In [6]:
anova_platforma = ols('Global_Sales ~ C(Platform)', data=data).fit()
anova_platforma_resultados = anova_lm(anova_platform)

# ANOVA 
anova_genero = ols('Global_Sales ~ C(Genre)', data=data).fit()
anova_genero_resultados = anova_lm(anova_genero)

# T-test: Global_Sales entre dos Platformas
platforma = data.groupby('Platform')['Global_Sales']
platforma_ttest = stats.ttest_ind(
    platforma.get_group('Wii'), 
    platforma.get_group('NES'), 
    equal_var=False
)

# T-test: Global_Sales entre dos Generos
generos = data.groupby('Genre')['Global_Sales']
genero_ttest = stats.ttest_ind(
    generos.get_group('Sports'), 
    generos.get_group('Platform'), 
    equal_var=False
)

In [7]:
anova_platforma_resultados

df        sum_sq    mean_sq          F         PR(>F)
C(Platform)     30.0   1447.579997  48.252667  20.600621  9.358994e-109
Residual     16509.0  38668.896207   2.342292        NaN            NaN

In [8]:
anova_genero_resultados

df        sum_sq    mean_sq          F        PR(>F)
C(Genre)     11.0    478.237226  43.476111  18.128282  1.698249e-36
Residual  16528.0  39638.238978   2.398248        NaN           NaN

In [9]:
platforma_ttest

TtestResult(statistic=-3.5596028734100416, pvalue=0.0005646379499878265, df=102.4930594881839)

In [10]:
genero_ttest

TtestResult(statistic=-3.8337098974636805, pvalue=0.00013207649516241808, df=1340.8405857773207)

### Resumen de Resultados

#### ANOVA:
Ventas globales por plataforma:
   - Estadístico F: 20.60
   - Valor p: \( 9.36 \times 10^{-109} \)
   - nterpretación: Hay diferencias significativas en las ventas globales entre las distintas plataformas.

Ventas globales por género:
   - Estadístico F: 18.13
   - Valor p: \( 1.70 \times 10^{-36} \)
   - Interpretación**: Hay diferencias significativas en las ventas globales entre los diferentes géneros de videojuegos.

### Prueba T:
Comparación de plataformas (Wii vs. NES):
   - Estadístico T: -3.56
   - Valor p: 0.00056
   - Interpretación: Existe una diferencia significativa en las ventas globales entre las plataformas Wii y NES.

Comparación de géneros (Deportes vs. Plataforma)**:
   - Estadístico T: -3.83
   - Valor p: 0.00013
   - Interpretación: Existe una diferencia significativa en las ventas globales entre los géneros de Deportes y Plataforma.

### Conclusión: 
   - Tanto el género como la plataforma influyen significativamente en las ventas globales.
